src/lib.rs

```diff
@@ -146,20 +146,41 @@ fn spectator_update(
 
         // translation
         {
-            let forward = if keys.pressed(KeyCode::W) { 1f32 } else { 0f32 };
-            let backward = if keys.pressed(KeyCode::S) { 1f32 } else { 0f32 };
-            let right = if keys.pressed(KeyCode::D) { 1f32 } else { 0f32 };
-            let left = if keys.pressed(KeyCode::A) { 1f32 } else { 0f32 };
+            let forward = if keys.pressed(KeyCode::KeyW) {
+                1f32
+            } else {
+                0f32
+            };
+
+            let backward = if keys.pressed(KeyCode::KeyS) {
+                1f32
+            } else {
+                0f32
+            };
+
+            let right = if keys.pressed(KeyCode::KeyD) {
+                1f32
+            } else {
+                0f32
+            };
+
+            let left = if keys.pressed(KeyCode::KeyA) {
+                1f32
+            } else {
+                0f32
+            };
```

Comment:
This is repeating the same pattern quiet often. You could use `condition as usize as f32` for an easy condition but I think that's a bit implicit so maybe something like

```rust
fn one_if_true(condition: bool) { if condition { 1.0 } else { 0.0 } }

let backward = one_if_true(keys.pressed(KeyCode::KeyS));
let right = one_if_true(keys.pressed(KeyCode::KeyD));
let left = one_if_true(keys.pressed(KeyCode::KeyA));
let forward = one_if_true(keys.pressed(KeyCode::KeyW));
```

would be more clean?

Reply:
I personally prefer the current implementation.